girder/models/file.py

```diff
@@ -93,23 +94,62 @@ def download(self, file, offset=0, headers=True, endByte=None,
         :type contentDisposition: str or None
         :type extraParameters: str or None
         """
+        if offset == 0:
+            events.trigger('model.file.download.before', info={'file': file})
+        else:
+            events.trigger('model.file.download.during', info={'file': file})
```

Comment:
IMO it would be even better to just have a single event that contained `startByte` and `endByte` fields.